pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/PersistentTopicsImpl.java

```diff
@@ -565,15 +573,15 @@ public void failed(Throwable throwable) {
             if (ex != null) {
                 futures.completeExceptionally(ex);
             } else {
-                messages.add(r);
-                List<CompletableFuture<Message>> futureMessages = Lists.newArrayList();
+                messages.addAll(r);
+                List<CompletableFuture<List<Message>>> futureMessages = Lists.newArrayList();
                 for (int i = 2; i <= numMessages; i++) {
                     futureMessages.add(peekNthMessage(destination, subName, i));
```

Comment:
should we break the loop if `messages.size() >= numMessages` because if client ask 100 messages and if msg is batch-msg with 1000 msgs in it then no need to peek next msg.?

Reply:
Had discussed it with Ludwig and came to a conclusion that Message in the context this API is a single entry stored in Bookkeeper.

Hence If the number of messages to peek is set to 3 then we print (3 * number of messages in batch) individually.

We can re-discuss this again if you think otherwise.